Don't mkdir existing directories when installing

Currently the install-* commands run mkdir directly when install-* commands are run, to ensure that the directory exists. mkdir raises an error if the directory already exists and this causes an abort of the build as more often than not, `/usr/local/lib" exists.

Some sort "ensure_exists" routine should be run instead.

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/32896125-don-t-mkdir-existing-directories-when-installing?utm_campaign=plugin&utm_content=tracker%2F283068&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F283068&utm_medium=issues&utm_source=github).
</bountysource-plugin>

@mgreter Sorry for the lack of information in the earlier comment and OP.

### System Information
OS: Ubuntu 14.04 64 bit
libsass: master branch

```
$ make --version
GNU Make 3.81
[snip]
This program built for x86_64-pc-linux-gnu
```

### How to Reproduce
I can reproduce by following running these commands to setup an environment:

```shell
cd /tmp
git clone https://github.com/sass/libsass libsass_make_check
cd libsass_make_check
make -j3
# Making sure the directory exists
sudo rm -rf /usr/local/include
sudo mkdir /usr/local/include
```

Subsequently, here's the output from the various make target commands

```shell
$ # Just checking for mkdir command...
$ sudo make -n install-headers
mkdir /usr/local/include
mkdir /usr/local/include/sass
install -v -m0644 "include/sass.h" "/usr/local/include/sass.h"
install -v -m0644 "include/sass2scss.h" "/usr/local/include/sass2scss.h"
install -v -m0644 "include/sass/base.h" "/usr/local/include/sass/base.h"
install -v -m0644 "include/sass/version.h" "/usr/local/include/sass/version.h"
install -v -m0644 "include/sass/values.h" "/usr/local/include/sass/values.h"
$ # And, failure.
$ sudo make install-headers
mkdir /usr/local/include
mkdir: cannot create directory ‘/usr/local/include’: File exists
make: *** [/usr/local/include] Error 1
$ # Incidentally, running the exact same make command again will work!
$ sudo make -n install-headers
install -v -m0644 "include/sass.h" "/usr/local/include/sass.h"
install -v -m0644 "include/sass2scss.h" "/usr/local/include/sass2scss.h"
install -v -m0644 "include/sass/base.h" "/usr/local/include/sass/base.h"
install -v -m0644 "include/sass/version.h" "/usr/local/include/sass/version.h"
install -v -m0644 "include/sass/values.h" "/usr/local/include/sass/values.h"
$ sudo make install-headers
install -v -m0644 "include/sass.h" "/usr/local/include/sass.h"
‘include/sass.h’ -> ‘/usr/local/include/sass.h’
install -v -m0644 "include/sass2scss.h" "/usr/local/include/sass2scss.h"
‘include/sass2scss.h’ -> ‘/usr/local/include/sass2scss.h’
install -v -m0644 "include/sass/base.h" "/usr/local/include/sass/base.h"
‘include/sass/base.h’ -> ‘/usr/local/include/sass/base.h’
install -v -m0644 "include/sass/version.h" "/usr/local/include/sass/version.h"
‘include/sass/version.h’ -> ‘/usr/local/include/sass/version.h’
install -v -m0644 "include/sass/values.h" "/usr/local/include/sass/values.h"
‘include/sass/values.h’ -> ‘/usr/local/include/sass/values.h’
install -v -m0644 "include/sass/context.h" "/usr/local/include/sass/context.h"
‘include/sass/context.h’ -> ‘/usr/local/include/sass/context.h’
install -v -m0644 "include/sass/functions.h" "/usr/local/include/sass/functions.h"
‘include/sass/functions.h’ -> ‘/usr/local/include/sass/functions.h’
```

--- a/Makefile
+++ b/Makefile
@@ -269,8 +269,7 @@ $(DESTDIR)$(PREFIX)/include/sass: | $(DESTDIR)$(PREFIX)/include
 	$(MKDIR) $(DESTDIR)$(PREFIX)/include/sass
 
 $(DESTDIR)$(PREFIX)/include/%.h: include/%.h \
-                                 | $(DESTDIR)$(PREFIX)/include \
-                                 $(DESTDIR)$(PREFIX)/include/sass
+                                 | $(DESTDIR)$(PREFIX)/include/sass
 	$(INSTALL) -v -m0644 "$<" "$@"
 
 install-headers: $(DESTDIR)$(PREFIX)/include/sass.h \

--- a/Makefile
+++ b/Makefile
@@ -230,13 +230,13 @@ debug-shared: shared
 lib:
 	$(MKDIR) lib
 
-lib/libsass.a: lib $(COBJECTS) $(OBJECTS)
+lib/libsass.a: $(COBJECTS) $(OBJECTS) | lib
 	$(AR) rcvs $@ $(COBJECTS) $(OBJECTS)
 
-lib/libsass.so: lib $(COBJECTS) $(OBJECTS)
+lib/libsass.so: $(COBJECTS) $(OBJECTS) | lib
 	$(CXX) -shared $(LDFLAGS) -o $@ $(COBJECTS) $(OBJECTS) $(LDLIBS)
 
-lib/libsass.dll: lib $(COBJECTS) $(OBJECTS) $(RCOBJECTS)
+lib/libsass.dll: $(COBJECTS) $(OBJECTS) $(RCOBJECTS) | lib
 	$(CXX) -shared $(LDFLAGS) -o $@ $(COBJECTS) $(OBJECTS) $(RCOBJECTS) $(LDLIBS) -s -Wl,--subsystem,windows,--out-implib,lib/libsass.a
 
 %.o: %.c
@@ -259,18 +259,18 @@ shared: $(SHAREDLIB)
 $(DESTDIR)$(PREFIX):
 	$(MKDIR) $(DESTDIR)$(PREFIX)
 
-$(DESTDIR)$(PREFIX)/lib: $(DESTDIR)$(PREFIX)
+$(DESTDIR)$(PREFIX)/lib: | $(DESTDIR)$(PREFIX)
 	$(MKDIR) $(DESTDIR)$(PREFIX)/lib
 
-$(DESTDIR)$(PREFIX)/include: $(DESTDIR)$(PREFIX)
+$(DESTDIR)$(PREFIX)/include: | $(DESTDIR)$(PREFIX)
 	$(MKDIR) $(DESTDIR)$(PREFIX)/include
 
-$(DESTDIR)$(PREFIX)/include/sass: $(DESTDIR)$(PREFIX)/include
+$(DESTDIR)$(PREFIX)/include/sass: | $(DESTDIR)$(PREFIX)/include
 	$(MKDIR) $(DESTDIR)$(PREFIX)/include/sass
 
 $(DESTDIR)$(PREFIX)/include/%.h: include/%.h \
-                                 $(DESTDIR)$(PREFIX)/include \
-                                 $(DESTDIR)$(PREFIX)/include/sass
+                                 | $(DESTDIR)$(PREFIX)/include \
+                                 | $(DESTDIR)$(PREFIX)/include/sass
 	$(INSTALL) -v -m0644 "$<" "$@"
 
 install-headers: $(DESTDIR)$(PREFIX)/include/sass.h \
@@ -282,15 +282,15 @@ install-headers: $(DESTDIR)$(PREFIX)/include/sass.h \
                  $(DESTDIR)$(PREFIX)/include/sass/functions.h
 
 $(DESTDIR)$(PREFIX)/lib/%.a: lib/%.a \
-                             $(DESTDIR)$(PREFIX)/lib
+                             | $(DESTDIR)$(PREFIX)/lib
 	@$(INSTALL) -v -m0755 "$<" "$@"
 
 $(DESTDIR)$(PREFIX)/lib/%.so: lib/%.so \
-                             $(DESTDIR)$(PREFIX)/lib
+                             | $(DESTDIR)$(PREFIX)/lib
 	@$(INSTALL) -v -m0755 "$<" "$@"
 
 $(DESTDIR)$(PREFIX)/lib/%.dll: lib/%.dll \
-                               $(DESTDIR)$(PREFIX)/lib
+                               | $(DESTDIR)$(PREFIX)/lib
 	@$(INSTALL) -v -m0755 "$<" "$@"
 
 install-static: $(DESTDIR)$(PREFIX)/lib/libsass.a
@@ -320,7 +320,7 @@ test_full: $(SASSC_BIN)
 test_probe: $(SASSC_BIN)
 	$(RUBY_BIN) $(SASS_SPEC_PATH)/sass-spec.rb -V 3.5 -c $(SASSC_BIN) --impl libsass --probe-todo $(LOG_FLAGS) $(SASS_SPEC_PATH)/$(SASS_SPEC_SPEC_DIR)
 
-clean-objects: lib
+clean-objects: | lib
 	-$(RM) lib/*.a lib/*.so lib/*.dll lib/*.la
 	-$(RMDIR) lib
 clean: clean-objects

--- a/Makefile
+++ b/Makefile
@@ -270,7 +270,7 @@ $(DESTDIR)$(PREFIX)/include/sass: | $(DESTDIR)$(PREFIX)/include
 
 $(DESTDIR)$(PREFIX)/include/%.h: include/%.h \
                                  | $(DESTDIR)$(PREFIX)/include \
-                                 | $(DESTDIR)$(PREFIX)/include/sass
+                                 $(DESTDIR)$(PREFIX)/include/sass
 	$(INSTALL) -v -m0644 "$<" "$@"
 
 install-headers: $(DESTDIR)$(PREFIX)/include/sass.h \

